Core/Graphics/Source/Graphics.cpp

```diff
@@ -64,12 +64,15 @@ namespace Babylon
 
     void Graphics::Impl::Resize(size_t width, size_t height)
     {
-        std::scoped_lock lock{m_bgfxState.Mutex};
+        m_resolution.width = width;
+        m_resolution.height = height;
+
         m_bgfxState.Dirty = true;
 
         auto& res = m_bgfxState.InitState.resolution;
-        res.width = static_cast<uint32_t>(width);
-        res.height = static_cast<uint32_t>(height);
+        auto ratio = GetHardwareScalingLevel();
+        res.width = static_cast<uint32_t>(m_resolution.width / ratio);
+        res.height = static_cast<uint32_t>(m_resolution.height / ratio);
```

Comment:
I would move all of this into a parameterless private `Resize` function and call that both here and from `SetHardwareScalingLevel`. Basically we are saying there are two different states that can affect final resolution: the view size, and the hardware scaling level. If either of those change, we need to do a `Resize` operation.

Reply:
Good suggestion, that prevents the current awkward way we call resize using the current resolution when the scaling level is changed.